src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -104,46 +107,90 @@ theorem gram_schmidt_pairwise_orthogonal (f : ι → E) :
 
 open submodule set order
 
-/-- `gram_schmidt` preserves span of vectors. -/
-lemma span_gram_schmidt [succ_order ι] [is_succ_archimedean ι] (f : ι → E) (c : ι) :
+lemma mem_span_gram_schmidt (f : ι → E) (i j : ι) (hij : i ≤ j) :
+  f i ∈ span 𝕜 (gram_schmidt 𝕜 f '' Iic j) :=
+begin
+  rw [gram_schmidt_def' 𝕜 f i],
+  refine submodule.add_mem _ (subset_span (mem_image_of_mem _ hij)) _,
+  apply submodule.sum_mem _ (λ k hk, _),
+  rw [orthogonal_projection_singleton],
+  apply smul_mem (span 𝕜 (gram_schmidt 𝕜 f '' Iic j)) _
+    (subset_span (set.mem_image_of_mem (gram_schmidt 𝕜 f) _)),
+  exact (le_of_lt (finset.mem_Iio.1 hk)).trans hij
+end
+
+lemma gram_schmidt_mem_span (f : ι → E) :
+  ∀ j i, i ≤ j → gram_schmidt 𝕜 f i ∈ span 𝕜 (f '' Iic j)
+| j := λ i hij, begin
+  rw [gram_schmidt_def 𝕜 f i],
+  refine submodule.sub_mem _ (subset_span (mem_image_of_mem _ hij))
+    (submodule.sum_mem _ (λ k hk, _)),
+  simp only [orthogonal_projection_singleton],
+  refine smul_mem _ _ _,
+  let hkj : k < j := (finset.mem_Iio.1 hk).trans_le hij,
+  refine span_mono _ (gram_schmidt_mem_span k k le_rfl),
+  exact set.image_subset f (Iic_subset_Iic.2 hkj.le)
+end
```

Comment:
The indentation of the begin-end block here looks very strange -- I think `begin` is supposed to be on its own line.